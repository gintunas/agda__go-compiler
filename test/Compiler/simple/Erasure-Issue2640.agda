-- Andreas, 2018-10-18, re #2757
-- Runtime-irrelevance analogue to issue #2640

-- {-# OPTIONS -v tc.lhs.unify:65 #-}
-- {-# OPTIONS -v tc.irr:20 #-}
-- {-# OPTIONS -v tc:30 #-}
-- {-# OPTIONS -v treeless:20 #-}

open import Common.Prelude

data D : (n : Nat) → Set where
  c : (m : Nat) → D m

test : (@0 n : Nat) → D n → Nat
test n (c m) = m

-- Should be rejected (projecting a forced argument).

main = printNat (test 142 (c _))

-- The generated Haskell program segfaults.
