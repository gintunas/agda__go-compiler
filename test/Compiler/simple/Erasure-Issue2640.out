EXECUTED_PROGRAM

ret > ExitFailure (-11)
